NishantAcharya/wwa
# Feature/base architecture

* Delete conflicting files

* Rename logic package to model

* Add first stage of the architecture

* Rename Main class to Application and add the base Game class

* Change Commands to interact with controllers instead of views; Change the run() of Game to static

* Add example classes, to show how you would use the architecture

* Remove previous models

* Test new git config

* Add an empty world class and extend a mock off of it

* Add a simple model class to the example

* Add support for arguments to the commands; Connect the World class to the architecture

* Remove public access to methods that should only be accessed in that package

* Add public access to the viewModel

* Change base ViewModel to an interface

* Add todo reminders for docs

* Add a message when an invalid command is sent

* Simplify example model

* Add WorldMock methods for the example

* Add a couple of example commands

* Rename base package to arch, and add a module package inside

* Add doc to command handler

* Add a way to navigate between modules

* Move command name constant to command class

* Fix up the navigator implementation

* Rename navigator interface to ModuleNavigator; Restrict some access modifiers

* Apply the navigator API to the controllers

* Replace String parameter in navigateTo() to an Enum

* Add docs for ICommand

* Add docs to Controller

* Add docs to Module, CommandHandler; Edit previous docs

* Remove assets file

* Add more to Module docs

* Add World docs

* Remove impl of the INavigator from the Module manager, and just send a base impl of it, instead of the whole Manager.

* Rename ModuleNavigator to IModuleNavigator

* Make properties that shouldn't be changed to final

* Rename Application to Main

* Add @see tags to some docs

* Add a safeguard if no ants exist

* Simplify creation of list in the WorldMock

* Add a way to create the World before the other objects are created

* Implement the starting manual command

* Remove completed TODO

* Fix example show ants command

* Move comment to docs

* Add a package for utils

* Enunciate the lack of utilities in this example

* Add functionality for modules to execute commands on their own when they are navigated to

* Add a way to end the game loop

* Provide an overloaded method for convenience in the command handler

* Move ModuleName enum outside of arch package, because it will have to be interacted with often, unlike the rest of the arch

* Remove World dependency from Controllers; Add an example of a service layer

* Move modules into module package
